contracts/tests/sbtc-token.test.ts

```diff
@@ -101,6 +101,197 @@ describe("sBTC token contract", () => {
+        CoreNodeEventType.ContractEvent
+      );
+      const [print] = printEvents;
+      const printData = cvToValue<{
+        topic: string;
+        txid: string;
+        voutIndex: bigint;
+        amount: bigint;
+      }>(print.data.value);
+      expect(printData).toStrictEqual({
+        topic: "completed-deposit",
+        bitcoinTxid: new Uint8Array(32).fill(0),
+        outputIndex: 0n,
+        amount: 1000n,
+        burnHash,
+        burnHeight: BigInt(burnHeight),
+        sweepTxid: new Uint8Array(32).fill(1),
+      });
+      const receipt1 = txOk(
+        token.transferMany({
+          recipients: [
+            {
+              amount: 100n,
+              sender: alice,
+              to: bob,
+              memo: null
+            },
+            {
+              amount: 100n,
+              sender: alice,
+              to: charlie,
+              memo: null
+            },
+          ],
+        }),
+        alice
+      );
+      expect(receipt1.value).toEqual(2n);
+      const receipt2 = rov(
+        token.getBalance({
+          who: bob,
+        }),
+        bob
+      );
+      expect(receipt2.value).toEqual(100n);
+      const receipt3 = rov(
+        token.getBalance({
+          who: charlie,
+        }),
+        charlie
+      );
+      expect(receipt3.value).toEqual(100n);
+    });
+
+    test("Mint & transfer multiple sbtc token, fail on first", () => {
+      const { burnHeight, burnHash } = getCurrentBurnInfo();
+      const receipt = txOk(
+        deposit.completeDepositWrapper({
+          txid: new Uint8Array(32).fill(0),
+          voutIndex: 0,
+          amount: 1000n,
+          recipient: alice,
+          burnHash,
+          burnHeight,
+          sweepTxid: new Uint8Array(32).fill(1),
+        }),
+        deployer
+      );
+      const printEvents = filterEvents(
+        receipt.events,
+        CoreNodeEventType.ContractEvent
+      );
+      const [print] = printEvents;
+      const printData = cvToValue<{
+        topic: string;
+        txid: string;
+        voutIndex: bigint;
+        amount: bigint;
+      }>(print.data.value);
+      expect(printData).toStrictEqual({
+        topic: "completed-deposit",
+        bitcoinTxid: new Uint8Array(32).fill(0),
+        outputIndex: 0n,
+        amount: 1000n,
+        burnHash,
+        burnHeight: BigInt(burnHeight),
+        sweepTxid: new Uint8Array(32).fill(1),
+      });
+      const receipt1 = txErr(
+        token.transferMany({
+          recipients: [
+            {
+              amount: 100n,
+              sender: alice,
+              to: bob,
+              memo: null
+            },
+            {
+              amount: 100n,
+              sender: alice,
+              to: charlie,
+              memo: null
+            },
+          ],
+        }),
+        bob
+      );
+      // Many-transfer errors start at 6, error happens at index 0
+      expect(receipt1.value).toEqual(errors.token.ERR_TRANSFER);
```

Comment:
I would add another case where the second transfer fails and the first succeeds so that we get the right error number (`errors.token.ERR_TRANSFER + 1`).